assets/js/modules/idea-hub/datastore/idea-state.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * `modules/idea-hub` data store: idea-state.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+
+const fetchPostUpdateIdeaStateStore = createFetchStore( {
+	baseName: 'updateIdeaState',
+	controlCallback: () => {
+		return API.set( 'modules', 'idea-hub', 'update-idea-state' );
+	},
+} );
```

Comment:
We need to validate and send `ideaState` as data to the `POST:update-idea-state` endpoint.

Reply:
@eugene-manuilov Ah yes, I forgot to pass the param here 🤦‍♂️   - thanks.
Regarding validation, I was a bit unsure because all the actions call `updateIdeaState` internally, and that then validates before calling the fetch. So I'm using `invariant` there to validate already.
Should I repeat / move the validation here using `validateParams`?

Reply:
yes, validation should happen in the create fetch datastore